staging/src/k8s.io/apiserver/pkg/endpoints/handlers/update.go

```diff
@@ -101,6 +102,30 @@ func UpdateResource(r rest.Updater, scope RequestScope, admit admission.Interfac
 				return newObj, mutatingAdmission.Admit(admission.NewAttributesRecord(newObj, oldObj, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Update, userInfo))
 			})
 		}
+		transformers = append(transformers, func(ctx context.Context, newObj, oldObj runtime.Object) (runtime.Object, error) {
+			if hasUID, err := hasUID(oldObj); err != nil {
+				return nil, err
+			} else if !hasUID && scope.Authorizer != nil {
+				createAttributes := authorizer.AttributesRecord{
+					User:            userInfo,
+					ResourceRequest: true,
+					Path:            req.URL.Path,
+					Verb:            "create",
+					APIGroup:        scope.Resource.Group,
+					APIVersion:      scope.Resource.Version,
+					Resource:        scope.Resource.Resource,
+					Subresource:     scope.Subresource,
+					Namespace:       namespace,
+					Name:            name,
+				}
+				if authorized, _, _ := scope.Authorizer.Authorize(createAttributes); authorized != authorizer.DecisionAllow {
+					// Instead of surfacing the error, just treat it as if the resource doesn't allow
+					// create on update, this way the Update loop will continue retrying.
```

Comment:
this response seems odd... what assumptions does this make about how storage Update handles not found errors? how does this surface to the user?

Reply:
I agree with this comment. Why not just fail and return an error stating the authz problem? The only benefit of retrying is if there is actually a race and we expect another component to create before our next iteration. But that case should be super super rare?

Reply:
The transformers on the objectInfo get evaluated before the update function checks the AllowCreateOnUpdate, so doing that would cause StatusForbidden to be returned to clients unauthorized to create making an update request to a nonexistent resource which doesn't allow create-on-update.

Reply:
We could have this unauthorized case return some special error that the Update function then checks and remembers that it was unauthorized before this check https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L527 if create-on-update is disabled

Reply:
> The transformers on the objectInfo get evaluated before the update function checks the AllowCreateOnUpdate

What if we hooked the create authz check into the createValidation passed down instead? That gets called *after* the storage has decided it wants to persist the input